src/providers/ipa/ipa_s2n_exop.c

```diff
@@ -1645,6 +1674,14 @@ static void ipa_s2n_get_user_done(struct tevent_req *subreq)
         state->attrs = attrs;
 
         if (attrs->response_type == RESP_USER_GROUPLIST) {
+
+            DEBUG(SSSDBG_TRACE_FUNC, "Received [%lu] groups in group list from "
+                                     "IPA Server\n", attrs->ngroups);
+
+            for (c = 0; c < attrs->ngroups; c++) {
+                DEBUG(SSSDBG_TRACE_FUNC, "[%s].\n", attrs->groups[c]);
+            }
+
```

Comment:
Maybe a

    if (DEBUG_IS_SET(SSSDBG_TRACE_FUNC)) {

around the whole debug block will help to prevent that the for-loop is always run even if there is nothing to log because the debug level is lower. This is done for loops at other places as well, see e.g. sdap_get_generic_ext_step().